### PR TITLE
Migrating the chart_pool controller to the new REST stack

### PR DESCRIPTION
**Pull request by ryanrath**
## Description
Migrating the 'chart_pool' controller / routes to the new REST Stack. 

## Motivation and Context
Because we don't want to support two rest stacks. 

## Tests performed
A suite of tests contained in `open_xdmod/modules/xdmod/integration_tests/lib/Controllers/ChartPoolTest.php`
It provides tests for the following: 
  - missing fields when /add is called  by a group of users.
    - fields: chart_id, chart_drill_details, chart_date_desc
    - users: cd, pi, usr
  - missing fields when /remove is called by a group of users
    - fields: chart_id
    - users: cd, pi, usr
  - the same chart cannot be added twice by various users
    - users: cd, pi, usr
  - that a various users should be able to add a chart and then remove it. 
  - that the public user cannot add / remove charts

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Code Refactor 

## Checklist:

- [X] My code follows the code style of this project as found in the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
-# Open XDMoD
+
+# Open XDMoD
```
Comment (jpwhite4) [inline README.md]: Please don't add blank lines to files that are unrelated to the functional changes to this pull request.

File: classes/ETL/Aggregator/pdoAggregator.php [hunk 2]
```diff
@@ -417,12 +417,11 @@ protected function verifyAggregationUnitTable($aggregationUnit)
             $tableFullName =  $utilitySchema . "." . $tableName;
             if ( false === $this->utilityEndpoint->tableExists($tableName, $utilitySchema) ) {
                 $this->logger->info("Table does not exist: '$tableFullName', skipping.");
-                continue;
             }
         } catch (PDOException $e) {
             $this->logAndThrowException(
                 "Error verifying aggregation unit table for '$aggregationUnit'",
-                array('exception' => $e, 'sql' => $sql)
+                array('exception' => $e)
```
Comment (jpwhite4) [inline classes/ETL/Aggregator/pdoAggregator.php]: It this change supposed to be included in this pull request? It appears to change code entirely unrelated to the chart_pool controller

File: classes/XDUser.php [hunk 3]
```diff
@@ -14,7 +14,7 @@
 class XDUser
 {
 
-    private $_pdo;                       // PDO Handle (set in __construct)
+    private $_pdo;                       //  PDO Handle (set in __construct)
```
Comment (jpwhite4) [inline classes/XDUser.php]: Please don't include this change in the pull request.

**Comment by ryanrath** [comment]
Closing this PR due to the extra git cruft that snuck in. I'll be opening a new one w/ a cleaned up set of changes.
